### PR TITLE
MGMT-10666 - Single Node expansion feature support flag

### PR DESCRIPTION
**Pull request by omertuc**
Starting OCP 4.11, adding workers to single node clusters is supported.

Currently, our UI shows a warning for all SNO clusters that says that
their expansion is not supported.

The UI needs to know when it should show the warning about users not
being able to expand a SNO cluster, and when it shouldn't (>=4.11).

To allow the UI to do that, the backend should report a feature support
flag for the 4.11 version, and this commit does that.

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
    - Deployed service and saw that the feature gets reported as expected in the API
- [ ] No tests needed

## Assignees

/cc @gamli75

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/featuresupport/support_levels_list.go [hunk 1]
```diff
@@ -105,4 +105,38 @@ var SupportLevelsList = models.FeatureSupportLevels{
...
+		// TODO: This list has been copied from 4.10 as-is just so I can add
+		// usage.SingleNodeExpansion as a supported feature for 4.11. This
+		// means I don't know know for sure whether all the features have the
+		// same support level in 4.11 as they had in 4.10, so take this list
+		// with a grain of salt and don't be afraid to change it if you know
+		// it should be different.
+		Features: []*models.FeatureSupportLevelFeaturesItems0{
+			// Supported
+			{
+				FeatureID:    usageNameToID(usage.SingleNodeExpansion),
```
Comment (gamli75) [inline internal/featuresupport/support_levels_list.go:119]: I think you should add it as unsupported to all other versions, I'm not sure what is the default behavior in the UI
@jkilzi what is the default?

File: internal/usage/consts.go [hunk 2]
```diff
@@ -37,4 +37,6 @@ const (
 	CPUArchitectureARM64 string = "arm64 architecture"
 	// The install config overrides specified by for the cluster
 	InstallConfigOverrides string = "Install Config Overrides"
+	// SNO expansion with workers
+	SingleNodeExpansion string = "Single Node Expansion"
```
[the remark this answers is not in the document]
Comment (tsorya) [reply internal/usage/consts.go:41]: Actually i just don't understand why we have this consts at all if there is swagger models

**Comment by gamli75** [comment]
@omertuc please review Igal's PR https://github.com/openshift/assisted-service/pull/3890 and rebase your PR

File: internal/featuresupport/support_levels_list.go (hunk 1 above)
Comment (omertuc) [reply internal/featuresupport/support_levels_list.go]: Sounds incredibly unnecessary & verbose. If a feature is not mentioned for a particular version, it should be assumed to be unsupported. Even if the UI is not prepared for this behavior, our service should do it automatically

Comment (omertuc) [reply internal/featuresupport/support_levels_list.go]: I'll resolve it in a separate PR

**Review by gamli75 — APPROVED** [review]

**Pull request merged**